docs/datasets.md

```diff
@@ -0,0 +1,683 @@
+
+**Usage:**
+```bash
+langstar dataset get <DATASET_ID> [OPTIONS]
+```
+
+**Arguments:**
+- `<DATASET_ID>` - UUID of the dataset
+
+**Options:**
+- `--json` - Output as JSON
+
+**Examples:**
+```bash
+# Get dataset details
+langstar dataset get 12345678-1234-1234-1234-123456789012
+
+# Get dataset as JSON
+langstar dataset get 12345678-1234-1234-1234-123456789012 --json
+```
+
+**Output:**
+```
+Dataset: my-qa-dataset
+  ID: 12345678-1234-1234-1234-123456789012
+  Type: kv
+  Description: Question-answer pairs for testing
+  Examples: 100
+  Sessions: 3
+  Created: 2025-11-20T10:00:00Z
+  Modified: 2025-11-28T12:00:00Z
+```
+
+### Update a Dataset
+
+Update dataset name or description.
+
+**Usage:**
+```bash
+langstar dataset update <DATASET_ID> [OPTIONS]
+```
+
+**Arguments:**
+- `<DATASET_ID>` - UUID of the dataset
+
+**Options:**
+- `--name <NEW_NAME>` - New name for the dataset
+- `--description <NEW_DESC>` - New description
+- `--json` - Output as JSON
+
+**Note:** At least one of `--name` or `--description` must be provided.
+
+**Examples:**
+```bash
+# Update dataset name
+langstar dataset update 12345678-1234-1234-1234-123456789012 \
+  --name "updated-qa-dataset"
+
+# Update description only
+langstar dataset update 12345678-1234-1234-1234-123456789012 \
+  --description "Updated: comprehensive Q&A pairs"
+
+# Update both name and description
+langstar dataset update 12345678-1234-1234-1234-123456789012 \
+  --name "production-qa-dataset" \
+  --description "Production-ready Q&A examples"
+```
+
+**Output:**
+```
+Dataset 12345678-1234-1234-1234-123456789012 updated successfully
+  Name: updated-qa-dataset
+  Description: Updated: comprehensive Q&A pairs
+```
+
+### Delete a Dataset
+
+Permanently delete a dataset and all its examples.
+
+**Usage:**
+```bash
+langstar dataset delete <DATASET_ID> [OPTIONS]
+```
+
+**Arguments:**
+- `<DATASET_ID>` - UUID of the dataset
+
+**Options:**
+- `-y, --yes` - Skip confirmation prompt (required for actual deletion)
+
+**Examples:**
+```bash
+# Attempt deletion (shows confirmation message)
+langstar dataset delete 12345678-1234-1234-1234-123456789012
+
+# Delete without confirmation
+langstar dataset delete 12345678-1234-1234-1234-123456789012 --yes
+```
+
+**Output:**
+```
+Deleted dataset 12345678-1234-1234-1234-123456789012
+```
+
+### Import Examples
+
+Import examples from a JSONL or CSV file into an existing dataset.
+
+**Usage:**
+```bash
+langstar dataset import <DATASET_ID> --file <FILE_PATH> [OPTIONS]
+```
+
+**Arguments:**
+- `<DATASET_ID>` - UUID of the dataset to import into
+
+**Options:**
+- `--file <PATH>` - Path to the file to import (required)
+- `--format <FORMAT>` - File format: `jsonl` or `csv` (auto-detected from extension)
+
+**Examples:**
+```bash
+# Import from JSONL file (format auto-detected)
+langstar dataset import 12345678-1234-1234-1234-123456789012 \
+  --file examples.jsonl
+
+# Import from CSV with explicit format
```

Comment:
Again what's the point? This is a useless feature: supporting people who want to misname their files